cabal-install/src/Distribution/Client/Init/Interactive/Command.hs

```diff
@@ -229,7 +239,7 @@ genTestTarget
     => InitFlags
     -> InstalledPackageIndex
     -> m (Maybe TestTarget)
-genTestTarget flags pkgs = initializeTestSuitePrompt flags >>= go
+genTestTarget flags pkgs = initializeTestSuitePrompt (flags {initializeTestSuite = NoFlag}) >>= go
```

Comment:
If i have `--lib --tests`, then this stomps all over the test flag.

Reply:
How so? Can you please elaborate? As I see it, `initializeTestSuite` must be set to `NoFlag` at that point, so it always prompt for testing, like you mentioned in https://github.com/haskell/cabal/issues/7410#issuecomment-849913926, `--tests` corresponds for a test suite package type, and `lib`/`--libandexe` will always ask for testing, hardcoding it to `NoFlag` basically does that. The only problem I see is when one passes only `--tests`, then it would ask when it should go straight through.

Reply:
I see what you mean now.

Reply:
Okay, fixed, turns out I over-complicated things and it was more straightforward than I thought.